BotProject/Templates/CSharp/Scripts/Node/botProjectDeploy.ts

```diff
@@ -0,0 +1,506 @@
+            }],
+            availableToOtherTenants: true,
+            replyUrls: [
+                'https://token.botframework.com/.auth/web/redirect'
+            ]
+        })
+        return createRes
+    }
+
+    private unpackObject(output: any) {
+        let unpakced: any = {}
+        for (const key in output) {
+            const objValue = output[key]
+            if (objValue.value) {
+                unpakced[key] = objValue.value
+            }
+        }
+        return unpakced
+    }
+
+    private async updateDeploymentJsonFile(settingsPath: string, client: ResourceManagementClient, resourceGroupName: string, deployName: string, appId: string, appPwd: string): Promise<any> {
+        const outputs = await client.deployments.get(resourceGroupName, deployName)
+        return new Promise((resolve, reject) => {
+            if (outputs.properties.outputs) {
+                const outputResult = outputs.properties.outputs
+                const applicatoinResult = {
+                    MicrosoftAppId: appId,
+                    MicrosoftAppPassword: appPwd
+                }
+                const outputObj = this.unpackObject(outputResult)
+
+                let result = {}
+                Object.assign(result, outputObj, applicatoinResult)
+
+                fs.writeFile(settingsPath, JSON.stringify(result, null, 4), (err) => {
+                    if (err) {
+                        reject(err)
+                    }
+                    resolve(result)
+                })
+            }
+            else {
+                resolve({})
+            }
+        })
+    }
+
+    private async writeToLog(data: any, logFile: string) {
+        return new Promise((resolve, reject) => {
+            fs.writeFile(logFile, JSON.stringify(data, null, 4), (err) => {
+                if (err) {
+                    reject(err)
+                }
+                resolve()
+            })
+        })
+    }
+
+    private async getFiles(dir: string): Promise<string[]> {
+        const dirents = await readdir(dir, { withFileTypes: true });
+        const files = await Promise.all(dirents.map((dirent) => {
+            const res = path.resolve(dir, dirent.name);
+            return dirent.isDirectory() ? this.getFiles(res) : res;
+        }));
+        return Array.prototype.concat(...files);
+    }
+
+    private async botPrepareDeploy(pathToDeploymentFile: string) {
+        return new Promise((resolve, reject) => {
+            const data = `[config]\nproject = BotProject.csproj`
+            fs.writeFile(pathToDeploymentFile, data, (err) => {
+                reject(err)
+            })
+        })
+    }
+
+    private async dotnetPublish(publishFolder: string, projFolder: string, botPath?: string) {
+        const projectPath = path.join(projFolder, 'BotProject.csproj')
+        await exec(`dotnet publish ${projectPath} -c release -o ${publishFolder} -v q`)
```

Comment:
To be consistent, you could consider using ARM to deploy rather than dotnet publish. The nice thing is that web app zip deploy through ARM will be useful the day we have other, non-dotnet languages, since it just requires a zip file. 

Here more info: https://docs.microsoft.com/en-us/azure/app-service/deploy-zip

Reply:
> To be consistent, you could consider using ARM to deploy rather than dotnet publish. The nice thing is that web app zip deploy through ARM will be useful the day we have other, non-dotnet languages, since it just requires a zip file.
> 
> Here more info: https://docs.microsoft.com/en-us/azure/app-service/deploy-zip

hi carloss, I use dotnet publish to produce an executable dotnet project, and then use the zip deploy you mentioned to publish that folder